_posts/2020-10-22-how-to-update-to-aws-provider-v3.adoc

```diff
@@ -0,0 +1,191 @@
+resources or data sources that are released from this point onwards, you will need to be compatible with version 3 of
+the AWS provider to be able to pull those changes in.
+
+Since this is a major version change, there are a number of backwards incompatible changes that were introduced. In
+order to be able to use version 3, you need to update the resources and data sources with backwards incompatible
+changes.
+
+For your own code, you can follow
+https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/version-3-upgrade[the official guide] on
+updating to AWS provider version 3. The rest of this guide will focus on updating references to Gruntwork modules and
+the Gruntwork Reference Architecture to be compatible with AWS provider version 3.
+
+
+[[deployment_walkthrough]]
+== Deployment walkthrough
+
+=== Updating references to Gruntwork Infrastructure as Code Library
+
+In order to take advantage of the Terraform AWS provider version 3, you need to update your references to the Gruntwork
+Infrastructure as Code Library to use a compatible version. The following table provides a summary of all the relevant
+Gruntwork AWS modules and the respective versions that are compatible with AWS provider version 3.
+
+[.exceptional]
+IMPORTANT: Gruntwork follows semantic versioning. In practice this means that version updates to the minor version are
+considered backwards incompatible releases for any version updates prior to 1.0.0 release. Make sure to read the release
+notes for the relevant modules any time you are updating minor versions!
+
+[cols="1,1h"]
+|===
+|IaC Package |Minimum version with AWS Provider v3 support
+
+|module-vpc
+|v0.9.0
+
+|module-asg
+|v0.10.0
+
+|module-data-storage
+|v0.15.0
+
+|module-server
+|v0.8.5
+
+|package-lambda
+|v0.8.1
+
+|package-sam
+|v0.2.1
+
+|module-cache
+|v0.9.4
+
+|package-messaging
+|v0.3.4
+
+|package-static-assets
+|v0.6.5
+
+|module-aws-monitoring
+|v0.22.1
+
+|package-openvpn
+|v0.11.0
+
+|module-security
+|v0.35.0
+
+|cis-compliance-aws
+|v0.6.0
+
+|module-ecs
+|v0.22.0
+
+|module-ci
+|v0.27.3
+
+|terraform-aws-eks
+|v0.22.1
+
+|module-load-balancer
+|v0.20.4
+
+|terraform-aws-couchbase
+|v0.3.0
+
+|package-zookeeper
+|v0.6.7
+
+|package-kafka
+|v0.6.3
+
+|package-elk
+|v0.6.0
+
+|terraform-aws-influx
+|v0.1.3
+
+|terraform-aws-vault
+|v0.13.11
+
+|terraform-aws-consul
+|v0.7.10
+
+|terraform-aws-nomad
+|v0.6.6
+
+|package-terraform-utilities
+|v0.2.1
+
+|===
+
+
+=== Updating the Gruntwork Reference Architecture to AWS Provider v3
+
+If you purchased the Gruntwork Reference Architecture, you will have a copy of the `infrastructure-live` and
+`infrastructure-modules` repositories that contain the infrastructure code for deploying the Reference Architecture. You
+will need to follow the steps to update the relevant code in `infrastructure-modules` to use a compatible version of the
+Gruntwork Infrastructure as Code Library, following the steps described above.
```

Comment:
```suggestion
`infrastructure-modules` repositories that contain the infrastructure code for deploying the Reference Architecture. You
will need to update the relevant code in `infrastructure-modules` to use a compatible version of the
Gruntwork Infrastructure as Code Library, as per the compatibility table above.
```

(consider making compatibility table a cross-link)

Reply:
Done in 97eb4617437e9bdd7270b9471260c4dde149764f